subxt/src/tx/tx_client.rs

```diff
@@ -50,59 +51,143 @@ impl<T: Config, C: OfflineClientT<T>> TxClient<T, C> {
+    /// up constructing either a v4 or v5 transaction. See [`Self::create_v4_unsigned`] or
+    /// [`Self::create_v5_bare`] if you'd like to explicitly create an unsigned transaction of a certain version.
+    pub fn create_unsigned<Call>(&self, call: &Call) -> Result<SubmittableTransaction<T, C>, Error>
     where
         Call: Payload,
     {
-        subxt_core::tx::create_unsigned(call, &self.client.metadata())
-            .map(|tx| SubmittableExtrinsic {
-                client: self.client.clone(),
-                inner: tx,
-            })
-            .map_err(Into::into)
+        let metadata = self.client.metadata();
+        let tx = match subxt_core::tx::suggested_version(&metadata)? {
+            TransactionVersion::V4 => subxt_core::tx::create_v4_unsigned(call, &metadata),
+            TransactionVersion::V5 => subxt_core::tx::create_v5_bare(call, &metadata),
+        }?;
+
+        Ok(SubmittableTransaction {
+            client: self.client.clone(),
+            inner: tx,
+        })
+    }
+
+    /// Creates a v4 unsigned (no signature or transaction extensions) transaction without submitting it.
+    ///
+    /// Prefer [`Self::create_unsigned()`] if you don't know which version to create; this will pick the
+    /// most suitable one for the given chain.
+    pub fn create_v4_unsigned<Call>(
+        &self,
+        call: &Call,
+    ) -> Result<SubmittableTransaction<T, C>, Error>
+    where
+        Call: Payload,
+    {
+        let metadata = self.client.metadata();
+        let tx = subxt_core::tx::create_v4_unsigned(call, &metadata)?;
+
+        Ok(SubmittableTransaction {
+            client: self.client.clone(),
+            inner: tx,
+        })
     }
 
-    /// Create a partial extrinsic.
+    /// Creates a v5 "bare" (no signature or transaction extensions) transaction without submitting it.
+    ///
+    /// Prefer [`Self::create_unsigned()`] if you don't know which version to create; this will pick the
+    /// most suitable one for the given chain.
+    pub fn create_v5_bare<Call>(&self, call: &Call) -> Result<SubmittableTransaction<T, C>, Error>
+    where
+        Call: Payload,
+    {
+        let metadata = self.client.metadata();
+        let tx = subxt_core::tx::create_v5_bare(call, &metadata)?;
+
+        Ok(SubmittableTransaction {
+            client: self.client.clone(),
+            inner: tx,
+        })
+    }
+
+    /// Create a partial transaction. Depending on the metadata, we might end up constructing either a v4 or
+    /// v5 transaction. See [`subxt_core::tx`] if you'd like to manually pick the version to construct
     ///
     /// Note: if not provided, the default account nonce will be set to 0 and the default mortality will be _immortal_.
     /// This is because this method runs offline, and so is unable to fetch the data needed for more appropriate values.
-    pub fn create_partial_signed_offline<Call>(
+    pub fn create_partial_offline<Call>(
```

Comment:
create_partial_signed_offline removed?

I guess one needs to call sign expliclity on it now?

Reply:
It's just calleed `create_partial_offline` now :)

The "signed" bit felt confusing because the thing being created is partial but it's definitely not signed yet!